### PR TITLE
Add methods to handle mcs addition in Expert filters

### PR DESCRIPTION
**Pull request by lilian-houdelet**

### CODE REVIEW

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java [hunk 1]
```diff
@@ -61,6 +62,21 @@ private static String getLoadFieldValue(FieldType field, Load load) {
         };
     }
 
+    private static String getShuntCompensatorFieldValue(FieldType field, ShuntCompensator shuntCompensator) {
+        return switch (field) {
+            case VOLTAGE_LEVEL_ID, COUNTRY, NOMINAL_VOLTAGE -> getVoltageLevelFieldValue(field, shuntCompensator.getTerminal().getVoltageLevel());
```
Comment (flomillot) [inline src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java:67]: Same syntax everywhere, not inline please

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java [hunk 2]
```diff
@@ -125,4 +141,18 @@ private static String getTerminalFieldValue(FieldType field, Terminal terminal)
...
+
+        return switch (field) {
+            case SHUNT_COMPENSATOR_TYPE -> susceptancePerSection > 0 ? "CAPACITOR" : "REACTOR";
+            case MAX_Q_AT_NOMINAL_V -> String.valueOf(voltageLevelPower * susceptancePerSection * shuntCompensator.getMaximumSectionCount());
+            case SWITCHED_ON_Q_AT_NOMINAL_V -> String.valueOf(voltageLevelPower * susceptancePerSection * shuntCompensator.getSectionCount());
+            case MAX_SUSCEPTANCE -> String.valueOf(susceptancePerSection * shuntCompensator.getMaximumSectionCount());
+            case SWITCHED_ON_MAX_SUSCEPTANCE -> String.valueOf(susceptancePerSection * shuntCompensator.getSectionCount());
+            default -> throw new PowsyblException(FIELD_AND_TYPE_NOT_IMPLEMENTED + " [" + field + ",terminal]");
+        };
+    }
```
Comment (flomillot) [inline src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java:157]: Where this part come from please, to let me check ?

**Review by flomillot — COMMENTED** [review]
Test to add

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java (hunk 2 above)
Comment (lilian-houdelet) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java:157]: It comes from network-modification-server in ShuntCompensatorModification.java (109 - 115)

**Review by dbraquart — CHANGES_REQUESTED** [review]
Code: renaming, and also 1 fix proposed for the issues below

Tests: all MCS fields have been tested. Issues:
- When we apply a filter (/evaluate rest api) on "maxQAtNominalV" we dont get the same card as in the spreadsheet
- Same with "SwitchedOnMaxQAtNominalV"

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java (hunk 2 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java]: yes but you missed the Math.abs

**Review by dbraquart — APPROVED** [review]
code: ok
tests: not done with last commit

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java [hunk 3]
```diff
@@ -128,6 +146,20 @@ private static String getTerminalFieldValue(FieldType field, Terminal terminal)
...
+    private static String getSectionBasedFieldValue(FieldType field, ShuntCompensator shuntCompensator) {
+        double susceptancePerSection = shuntCompensator.getModel(ShuntCompensatorLinearModel.class).getBPerSection();
+        double qAtNominalV = Math.pow(shuntCompensator.getTerminal().getVoltageLevel().getNominalV(), 2) * Math.abs(susceptancePerSection);
+
+        return switch (field) {
+            case SHUNT_COMPENSATOR_TYPE -> susceptancePerSection > 0 ? "CAPACITOR" : "REACTOR";
+            case MAX_Q_AT_NOMINAL_V -> String.valueOf(qAtNominalV * shuntCompensator.getMaximumSectionCount());
+            case SWITCHED_ON_Q_AT_NOMINAL_V -> String.valueOf(qAtNominalV * shuntCompensator.getSectionCount());
+            case MAX_SUSCEPTANCE -> String.valueOf(susceptancePerSection * shuntCompensator.getMaximumSectionCount());
+            case SWITCHED_ON_MAX_SUSCEPTANCE -> String.valueOf(susceptancePerSection * shuntCompensator.getSectionCount());
```
Comment (flomillot) [inline src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java:158]: Why `SWITCHED_ON_MAX_SUSCEPTANCE` and not `SWITCHED_ON_SUSCEPTANCE` ?

**Review by flomillot — CHANGES_REQUESTED** [review]
Missing tests on VOLTAGE_LEVEL_ID in StringExpertRuleTest

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java (hunk 3 above)
Comment (lilian-houdelet) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java:158]: the key in grid study is "SwitchedOnMaxSusceptance", so I used it like that

Comment (flomillot) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java:158]: `    "SwitchedOnMaxSusceptance": "Switch-on susceptance",`
Looking at the translation, I think it's a mistake.

**Review by flomillot — APPROVED** [review]

**Pull request merged**